src/main/java/games/strategy/net/ServerMessenger.java

```diff
@@ -247,20 +248,24 @@ public void notifyPlayerLogin(final String uniquePlayerName, final String ip, fi
       if (isLobby()) {
         final String realName = uniquePlayerName.split(" ")[0];
         if (!liveMutedUsernames.contains(realName)) {
-          final long muteTill = new MutedUsernameController().getUsernameUnmuteTime(realName);
-          if (muteTill != -1 && muteTill <= System.currentTimeMillis()) {
-            // Signal the player as muted
-            liveMutedUsernames.add(realName);
-            scheduleUsernameUnmuteAt(realName, Instant.ofEpochMilli(muteTill));
-          }
+          final Optional<Instant> muteTill = new MutedUsernameController().getUsernameUnmuteTime(realName);
+          muteTill.ifPresent(instant -> {
+            if (!Instant.EPOCH.equals(instant) && Instant.now().isAfter(instant)) {
+              // Signal the player as muted
```

Comment:
If an empty `Optional` means an infinite mute, won't this incorrectly show the player is _not_ muted in such a case?

Also, shouldn't the `isAfter()` operands be reversed?  That is, `instant.isAfter(Instant.now())`?  I realize the original code has `muteTill <= System.currentTimeMillis()`, but that doesn't seem right (or I'm just up too late :smile:).  If the mute expiration time is in the past, why would we signal the player as muted?

If you decide to use an empty `Optional` to signal no mute, as suggested above, I think this can be simplified to

```java
final Optional<Instant> muteTill = new MutedUsernameController().getUsernameUnmuteTime(realName);
muteTill.ifPresent(instant -> {
  if (instant.isAfter(Instant.now())) {
    // Signal the player as muted
    liveMutedUsernames.add(realName);
    scheduleUsernameUnmuteAt(realName, instant);
  }
});
```

Reply:
Yeah, it really doesn't seem right, I was confused as well, but for me it was too late to bother so I just kept the same behaviour.
Seems like this bug has been present ever since the lobby has moderation features:
https://github.com/RoiEXLab/triplea/commit/225dcdd74340cff7263d1a3764c45d59315c319f#diff-789d2869e390c4eec6be2ad040841375R308

Reply:
Yeah, I guess no one noticed because this code only kicks in when a muted user is logging in, not when an admin actually mutes them.  I wonder if this means all a muted user has to do to get around the mute is to log out and log back in?